src/test/javascript/spec/component/example-modeling/example-modeling-submission.component.spec.ts

```diff
@@ -0,0 +1,280 @@
+import { routes } from 'app/exercises/modeling/manage/modeling-exercise.route';
+import { ArtemisExampleModelingSubmissionRoutingModule } from 'app/exercises/modeling/manage/example-modeling/example-modeling-submission.route';
+import { TutorParticipationService } from 'app/exercises/shared/dashboards/tutor/tutor-participation.service';
+import { ExerciseService } from 'app/exercises/shared/exercise/exercise.service';
+import { ModelingAssessmentService } from 'app/exercises/modeling/assess/modeling-assessment.service';
+
+describe('Example Modeling Submission Component', () => {
+    let comp: ExampleModelingSubmissionComponent;
+    let fixture: ComponentFixture<ExampleModelingSubmissionComponent>;
+    let debugElement: DebugElement;
+    let service: ExampleSubmissionService;
+    let alertService: AlertService;
+    let router: Router;
+    let route: ActivatedRoute;
+
+    const participation = new StudentParticipation();
+    participation.exercise = new ModelingExercise(UMLDiagramType.ClassDiagram, undefined, undefined);
+    participation.id = 1;
+    const submission = { id: 20, submitted: true, participation } as ModelingSubmission;
+
+    const exampleSubmission: ExampleSubmission = {
+        submission,
+    };
+
+    const exercise = {
+        diagramType: UMLDiagramType.ClassDiagram,
+    } as ModelingExercise;
+
+    const mockFeedbackWithReference = {
+        text: 'FeedbackWithReference',
+        referenceId: 'relationshipId',
+        reference: 'reference',
+        credits: 30,
+        correctionStatus: 'CORRECT',
+    } as Feedback;
+    const mockFeedbackWithoutReference = { text: 'FeedbackWithoutReference', credits: 30, type: FeedbackType.MANUAL_UNREFERENCED } as Feedback;
+    const mockFeedbackInvalid = { text: 'FeedbackInvalid', referenceId: '4', reference: 'reference', correctionStatus: FeedbackCorrectionErrorType.INCORRECT_SCORE };
+
+    beforeEach(() => {
+        route = {
+            snapshot: {
+                paramMap: convertToParamMap({ exerciseId: '22', exampleSubmissionId: '35' }),
+                queryParamMap: convertToParamMap({ readOnly: 0, toComplete: 0 }),
+            },
+        } as ActivatedRoute;
+
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, RouterTestingModule.withRoutes([routes[0]]), MockModule(ArtemisExampleModelingSubmissionRoutingModule)],
+            declarations: [ExampleModelingSubmissionComponent, MockComponent(ModelingEditorComponent), MockPipe(ArtemisTranslatePipe)],
+            providers: [MockProvider(ChangeDetectorRef), { provide: ActivatedRoute, useValue: route }],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(ExampleModelingSubmissionComponent);
+                comp = fixture.componentInstance;
+                debugElement = fixture.debugElement;
+                service = debugElement.injector.get(ExampleSubmissionService);
```

Comment:
Thanks for the review. According to the [Angular docs](https://angular.io/guide/testing-components-scenarios#get-injected-services) getting it this way ensures it is the same service instance that is actually injected into the component so it may be better to keep it.